samples/AzureFunctionsApp/Entities/User.cs

```diff
@@ -0,0 +1,171 @@
+using Microsoft.DurableTask.Client.Entities;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace AzureFunctionsApp.Entities;
+
+public record User(string Name, int Age);
+
+public record UserUpdate(string? Name, int? Age);
+
+/// <summary>
+/// This sample demonstrates how to bind to <see cref="TaskEntityContext"/> as well as dispatch to orchestrations.
+/// </summary>
+public class UserEntity : TaskEntity<User>
+{
+    readonly ILogger logger;
+
+    public UserEntity(ILogger<UserEntity> logger)
+    {
+        this.logger = logger;
+    }
+
+    public void Set(User user)
+    {
+        User previous = this.State;
+        this.State = user;
+        this.logger.LogInformation("User {Id} set {Old} -> {New}", this.Context.Id.Key, previous, this.State);
+    }
+
+    public void Update(UserUpdate update)
+    {
+        (string n, int a) = (update.Name ?? this.State.Name, update.Age ?? this.State.Age);
+        User previous = this.State;
+        this.State = previous with { Name = n, Age = a };
+
+        this.logger.LogInformation("User {Id} updated {Old} -> {New}", this.Context.Id.Key, previous, this.State);
+    }
+
+    /// <summary>
+    /// Starts a <see cref="Greeting.GreetingOrchestration(TaskOrchestrationContext, Greeting.Input)"/>.
+    /// </summary>
+    /// <param name="context">The context object. This will allow for calling orchestrations.</param>
+    /// <param name="message">
+    /// The optional message. By using a default parameter supplying input to this operation is optional.
+    /// </param>
+    public void Greet(TaskEntityContext context, string? message = null)
+    {
+        if (this.State.Name is null)
+        {
+            this.logger.LogError("User is not in a valid state for a greet operation.");
+            throw new InvalidOperationException("User has not been initialized.");
+        }
+
+        // Get access to TaskEntityContext by adding it as a parameter. Can be with or without an input parameter.
+        // Order does not matter.
+        Greeting.Input input = new(this.State.Name, this.State.Age) { CustomMessage = message };
+        context.ScheduleNewOrchestration(nameof(Greeting.GreetingOrchestration), input);
+
+        // When using TaskEntity<TState>, the TaskEntityContext can also be accessed via this.Context.
+        // this.Context.ScheduleNewOrchestration(nameof(Greeting.GreetingOrchestration), input);
+    }
+
+    [Function(nameof(User))]
+    public Task DispatchAsync([EntityTrigger] TaskEntityDispatcher dispatcher)
+    {
+        // Can dispatch to a TaskEntity<TState> by passing a instance.
+        return dispatcher.DispatchAsync(this);
+    }
+
+    protected override User InitializeState(TaskEntityOperation entityOperation)
+    {
+        // No parameterless constructor, must initialize state manually.
+        return new(null!, -1);
+    }
+}
+
+public static class UserApis
+{
+    [Function("PutUser")]
```

Comment:
It may be helpful to start this section with a set of examples for the REST invocations, e.g.

```
PUT    user/xyz?name=aang&age=10
GET    user/xyz
DELETE user/xyz
PATCH  user/xyz?age=12
POST   user/xyz/greet?message=hello
```